frontend/src/pages/Posts.jsx

```
@@ -247,18 +274,86 @@ const Posts = () => {
+                onChange={(e) => setMedium(e.target.value)}
+                className="w-full pl-10 pr-10 py-2 border border-gray-300 rounded-full focus:ring-2 focus:ring-indigo-500 focus:border-transparent transition-all duration-300"
+              >
+                <option value="">Select Medium</option>
+                <option value="BANGLA">Bangla</option>
+                <option value="ENGLISH">English</option>
+              </select>
+              {medium && (
+                <button 
+                  onClick={() => setMedium("")}
+                  className="absolute inset-y-0 right-0 pr-3 flex items-center"
+                >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)